### ISSUE TITLE
`cargo clippy --explain LINT` only works on stable versions, regardless of which lint is passed

### ISSUE THREAD

**Issue by hashcatHitman** [rust-lang/rust-clippy#17620]
### Summary

When using any toolchain newer than stable (beta/nightly), using `cargo clippy --explain LINT` to get documentation about any lint fails with the message `unknown lint: LINT` (where LINT in each case is a placeholder, to be clear). There's a similar issue #13822, but the discussion seems to suggest that had more to do with the lint in question being too new. These lints are old.

### Reproducer

```bash
cargo +stable clippy --explain allow_attributes
cargo +beta clippy --explain allow_attributes
cargo +nightly clippy --explain allow_attributes
cargo +stable clippy --explain absurd_extreme_comparisons
cargo +beta clippy --explain absurd_extreme_comparisons
cargo +nightly clippy --explain absurd_extreme_comparisons
cargo +stable clippy --explain toplevel_ref_arg
cargo +beta clippy --explain toplevel_ref_arg
cargo +nightly clippy --explain toplevel_ref_arg
```

Current output:

All of the `+stable` work, but none of the `+beta` or `+nightly` work. So you will get:

```text
<DOCS FOR allow_attributes>
unknown lint: allow_attributes
unknown lint: allow_attributes
<DOCS FOR absurd_extreme_comparisons>
unknown lint: absurd_extreme_comparisons
unknown lint: absurd_extreme_comparisons
<DOCS FOR toplevel_ref_arg>
unknown lint: toplevel_ref_arg
unknown lint: toplevel_ref_arg
```

Desired output:

All of them should work.

```text
<DOCS FOR allow_attributes>
<DOCS FOR allow_attributes>
<DOCS FOR allow_attributes>
<DOCS FOR absurd_extreme_comparisons>
<DOCS FOR absurd_extreme_comparisons>
<DOCS FOR absurd_extreme_comparisons>
<DOCS FOR toplevel_ref_arg>
<DOCS FOR toplevel_ref_arg>
<DOCS FOR toplevel_ref_arg>
```

### Version

```text
cargo +stable clippy -V
clippy 0.1.98 (88d9e12ae1 2026-08-18)

cargo +beta clippy -V
clippy 0.1.99 (f47d5bb136 2026-08-17)

cargo +nightly clippy -V
clippy 0.1.99 (c656540d64 2026-08-21)

cargo clippy -V
clippy 0.1.99 (c656540d64 2026-08-21)

rustc -Vv
rustc 1.100.0-nightly (c656540d6 2026-08-21)
binary: rustc
commit-hash: c656540d6467dee1381f0cbd882412d6bd1cd5ae
commit-date: 2026-08-21
host: x86_64-unknown-linux-gnu
release: 1.100.0-nightly
LLVM version: 23.1.0
```

**Comment by arcusbuilds** [comment]
I would like to work on it!

**Comment by hashcatHitman** [comment]
> I would like to work on it!

Great! You can comment `@rustbot claim` to assign yourself to the issue so others know you're currently working on it.

**Comment by Jarcho** [comment]
I accidentally broke this in #13084
